src/chart/helper/multipleGraphEdgeHelper.js

```diff
@@ -0,0 +1,221 @@
+ * @param link
+ * @param seriesModel
+ * @returns {string}
+ */
+var getKeyOfLinks = function (link, seriesModel) {
+    return [seriesModel.uid, link.source, link.target].join('>');
+};
+
+/**
+ * get opposite key
+ * 获取反向的key
+ * @param key
+ * @returns {string}
+ */
+var getOppositeKey = function (key) {
+    var keys = key.split('>');
+
+    return [keys[0], keys[2], keys[1]].join('>');
+};
+
+/**
+ * set linkMap with key
+ * @param link
+ * @param seriesModel
+ * @param index
+ */
+var setLinkToMap = function (link, seriesModel, index) {
+    var key = getKeyOfLinks(link, seriesModel);
+    var hasOppositeLinks = linkMap[getOppositeKey(key)];
+    // set direction
+    if (hasOppositeLinks && linkMap[key] && !linkMap[key].isForward) {
+        linkMap[getOppositeKey(key)].isForward = true;
+    }
+
+    linkMap[key] = linkMap[key] || [];
+    linkMap[key].push(index);
+};
+
+/**
+ * get linkMap with key
+ * @param link
+ * @param seriesModel
+ * @param index
+ */
+var getLinkFromMap = function (link, seriesModel) {
+    var key = getKeyOfLinks(link, seriesModel);
+    return linkMap[key];
+};
+
+/**
+ * calculate all cases total length
+ * @param link
+ * @param seriesModel
+ * @returns {number}
+ */
+var getTotalLengthBetweenNodes = function (link, seriesModel) {
+    var len = getLinkMapLengthWithKey([seriesModel.uid, link.source, link.target].join('>'));
+    var lenV = getLinkMapLengthWithKey([seriesModel.uid, link.target, link.source].join('>'));
+
+    return len + lenV;
+};
+
+/**
+ *
+ * @param key
+ */
+var getLinkMapLengthWithKey = function (key) {
+    return linkMap[key] ? linkMap[key].length : 0;
+};
+
+/**
+ * Count the number of edges between the same two points, used to obtain the curvature table and the parity of the edge
+ * @param edges
+ * @param seriesModel
+ * @param graph
+ */
+export function calculateMutilEdges(edges, seriesModel, graph) {
+    if (!getAutoCurvenessParams(seriesModel)) {
+        return;
+    }
+    // Hang on this object 4 dispose
+    curvenessList = graph.curvenessList || [];
+    linkMap = graph.linkMap = graph.linkMap || {};
+
+
+    for (var i = 0; i < edges.length; i++) {
+        var link = edges[i];
+        var source = link.source;
+        var target = link.target;
+        if (!source || !target) {
+            continue;
+        }
+        setLinkToMap(link, seriesModel, i);
+    }
+
+    // calc the array of curvenessList
+    createCurveness(seriesModel);
+}
+
+/**
+ * Set curvature for link
+ * @param link
+ * @param seriesModel
+ * @param index
+ */
+export function setCurvenessForLink(link, seriesModel, index) {
+    if (!getAutoCurvenessParams(seriesModel)) {
+        return;
+    }
+
+    var linkArray = getLinkFromMap(link, seriesModel);
+    if (!linkArray) {
+        return;
+    }
+
+    var linkIndex = linkArray.findIndex(function (it) {
+        return it === index;
+    });
```

Comment:
Currently source code of echarts is targeted at es3 and no polyfill yet. So `findIndex` should better not be used.
The code snippet below could be a substitution:
```js
    var linkIndex = -1;
    for (var i = 0; i < linkArray.length; i++) {
        if (linkArray[i] === index) {
            linkIndex = i;
            break;
        }
    }
```